cluster/cluster.go

```diff
@@ -33,6 +33,26 @@ import (
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+// ClusterPeer helps consumers gather information about the peer(s) in the cluster
+type ClusterPeer interface {
+	// Name returns the unique identifier of this peer in the cluster.
+	Name() string
+	// Address returns the IP address of this peer in the cluster.
+	Address() string
+	// Status returns a status string representing the peer state.
+	Status() string
+	// Peers returns the peer nodes in the cluster.
+	Peers() []Node
+}
+
+// Node interface that represents node peers in a cluster
+type Node interface {
```

Comment:
Maybe `ClusterMember` could be a better naming, given it represent a member of cluster?

Reply:
I like this, thanks!